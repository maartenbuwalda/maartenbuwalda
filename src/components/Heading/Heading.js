import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const Wrapper = styled.div`
  transition: background-color .3s;
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 3rem;
  background-color: ${props => props.color.hex.colorMutedRegular};
  margin-bottom: 3rem;
`

const LinkWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 1rem;

  a {
    padding: 1rem;
    transition: color .3s;
    text-decoration: none;
    color: ${props => props.color.hex.colorMutedLight};
  }
`

const Heading = ({ color }) => (
  <Wrapper color={color}>
    <LinkWrapper color={color}>
      <Link to="/">home</Link>
      <Link to="/hello">hello</Link>
    </LinkWrapper>
  </Wrapper>
)

Heading.propTypes = {
  color: PropTypes.object,
}

export default Heading
