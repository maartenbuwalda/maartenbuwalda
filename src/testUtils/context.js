const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
