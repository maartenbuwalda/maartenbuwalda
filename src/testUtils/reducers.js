export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
