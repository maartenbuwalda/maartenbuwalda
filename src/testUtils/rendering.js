import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}

export const connectedComponentRenders = component => {
  const middlewares = []
  const mockStore = configureStore(middlewares)
  const initialState = {}
  const store = mockStore(initialState)
  
  it('renders correctly', () => {
    const shallowed = shallow(component, store)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}
