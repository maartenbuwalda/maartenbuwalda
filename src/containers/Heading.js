import Heading from '../components/Heading'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'

const mapStateToProps = state => {
  return {
    color: state.color,
  }
}

export default withRouter(connect(mapStateToProps)(Heading))
