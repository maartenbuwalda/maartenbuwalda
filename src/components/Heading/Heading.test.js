import React from 'react'
import Heading from './index'
import { componentRenders } from '../../testUtils/rendering'

describe('Heading component', () => {
  const component = <Heading />
  componentRenders(component)
})
