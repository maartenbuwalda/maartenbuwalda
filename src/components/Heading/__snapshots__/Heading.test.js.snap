// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component renders correctly 1`] = `
<styled.div>
  <styled.div>
    <Link
      replace={false}
      to="/"
    >
      home
    </Link>
    <Link
      replace={false}
      to="/hello"
    >
      about
    </Link>
  </styled.div>
</styled.div>
`;
